heron/packing/tests/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPackingTest.java

```diff
@@ -0,0 +1,540 @@
+  public void testDefaultResources() throws Exception {
+    int numContainers = 2;
+    int spoutParallelism = 4;
+    int boltParallelism = 3;
+    int totalInstances = spoutParallelism + boltParallelism;
+
+    // Set up the topology and its config
+    com.twitter.heron.api.Config topologyConfig = new com.twitter.heron.api.Config();
+    topologyConfig.put(com.twitter.heron.api.Config.TOPOLOGY_STMGRS, numContainers);
+
+    // No explicit resources required
+    TopologyAPI.Topology topologyNoExplicitResourcesConfig =
+        getTopology(spoutParallelism, boltParallelism, topologyConfig);
+
+    PackingPlan packingPlanNoExplicitResourcesConfig =
+        getResourceCompliantRRPackingPlan(topologyNoExplicitResourcesConfig);
+
+    Assert.assertEquals(packingPlanNoExplicitResourcesConfig.containers.size(), 2);
+
+    long totalRam = (totalInstances + HERON_INTERNAL_CONTAINERS)
+        * instanceRamDefault
+        + (long) ((DEFAULT_CONTAINER_PADDING / 100.0)
+        * totalInstances * instanceRamDefault);
+
+    Assert.assertEquals(packingPlanNoExplicitResourcesConfig.resource.ram, totalRam);
+
+    double totalCpu = Math.round(4 * instanceCpuDefault
+        + (DEFAULT_CONTAINER_PADDING / 100.0) * 4 * instanceCpuDefault)
+        + Math.round(3 * instanceCpuDefault
+        + (DEFAULT_CONTAINER_PADDING / 100.0) * 3 * instanceCpuDefault)
+        + instanceCpuDefault;
+
+    Assert.assertEquals((long) packingPlanNoExplicitResourcesConfig.resource.cpu, (long) totalCpu);
+
+    long totalDisk = (totalInstances + HERON_INTERNAL_CONTAINERS)
+        * instanceDiskDefault
+        + (long) ((DEFAULT_CONTAINER_PADDING / 100.0)
+        * totalInstances * instanceDiskDefault);
+
+    Assert.assertEquals(packingPlanNoExplicitResourcesConfig.resource.disk, totalDisk);
+  }
+
+  /**
+   * Test the scenario where the max container size is the default and padding is configured
+   */
+  @Test
+  public void testDefaultContainerSizeWithPadding() throws Exception {
+    int numContainers = 2;
+    int spoutParallelism = 4;
+    int boltParallelism = 3;
+    int padding = 50;
+    int totalInstances = spoutParallelism + boltParallelism;
+    // Set up the topology and its config
+    com.twitter.heron.api.Config topologyConfig = new com.twitter.heron.api.Config();
+    topologyConfig.put(com.twitter.heron.api.Config.TOPOLOGY_STMGRS, numContainers);
+    topologyConfig.setContainerPaddingPercentage(padding);
+    TopologyAPI.Topology topology =
+        getTopology(spoutParallelism, boltParallelism, topologyConfig);
+
+    PackingPlan packingPlan =
+        getResourceCompliantRRPackingPlan(topology);
+
+    Assert.assertEquals(packingPlan.containers.size(), 2);
+
+    long totalRam = (totalInstances + HERON_INTERNAL_CONTAINERS)
+        * instanceRamDefault
+        + (long) ((padding / 100.0) * totalInstances * instanceRamDefault);
+
+    Assert.assertEquals(packingPlan.resource.ram, totalRam);
+
+    double totalCpu = Math.round(4 * instanceCpuDefault
```

Comment:
They are not spoutParallelism and BoltParallelism. In this test, the first container will contain 2 spouts and 2 bolts (4) and the second one (2 spouts and 1 bolt) so it cannot be replaced.